### PR TITLE
Use new Symbol APIs in Profiler example

### PR DESCRIPTION
**Pull request by andrewfayres**
## Description ##
As part of the effort to update models to the new inference API we're also moving to a new folder. This example isn't an inference one but still needs to be moved.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

@lanking520 @nswamy

### CODE REVIEW

**Comment by andrewfayres** [comment]
Odd, I know I updated these earlier. I'll take a look when I get home.

**Comment by nswamy** [comment]
@andrewfayres can you please add tests to this?

File: scala-package/examples/src/test/scala/org/apache/mxnetexamples/infer/profiler/ProfilerTest.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.io.File
+
+import org.apache.mxnet.Profiler
+import org.apache.mxnet.Context
+
+/**
+  * Integration test for imageClassifier example.
+  * This will run as a part of "make scalatest"
+  */
+class ProfilerTest extends FunSuite with BeforeAndAfterAll {
```
Comment (yzhliu) [inline scala-package/examples/src/test/scala/org/apache/mxnetexamples/infer/profiler/ProfilerTest.scala]: better to move tests to `test` package,  use `before{` for `Profiler.profilerSetState("run")`, `after{` for `Profiler.profilerSetState("stop")`, and move each `testXXX` in different `test{` scope: http://www.scalatest.org/getting_started_with_fun_suite

File: scala-package/examples/src/test/scala/org/apache/mxnetexamples/infer/profiler/ProfilerSuite.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (lanking520) [inline scala-package/examples/src/test/scala/org/apache/mxnetexamples/infer/profiler/ProfilerSuite.scala]: Why infer, it should live in `mxnetexamples` at least

Comment (andrewfayres) [reply scala-package/examples/src/test/scala/org/apache/mxnetexamples/infer/profiler/ProfilerSuite.scala]: You're right. Stuck it in the wrong place.

**Pull request merged**